hydra/_internal/utils.py

```diff
@@ -290,6 +292,8 @@ def get_args_parser() -> argparse.ArgumentParser:
     shell = "SHELL_NAME"
     install_cmd = 'eval "$({} -sc install={})"'.format(_get_exec_command(), shell)
     uninstall_cmd = 'eval "$({} -sc uninstall={})"'.format(_get_exec_command(), shell)
+    fish_install_cmd = "{} -sc install=fish | source".format(_get_exec_command())
+    fish_uninstall_cmd = "{} -sc uninstall=fish | source".format(_get_exec_command())
```

Comment:
is the eval command above not working for fish?

Reply:
No. Fish's eval command only works for a single line. [fish discussion](https://github.com/fish-shell/fish-shell/issues/3993)

Reply:
The evaluated expression is a single line, isn't it?
Does the `| source` approach works for bash?

Reply:
The expression was multiple lines. If we use the way like bash, it would contain more lines. Bash `source` doesn't support read from stdin so this approach doesn't work for it.